web/src/components/Login.js

```diff
@@ -1,39 +1,46 @@
-import React, { useState } from 'react';
-import { useNavigate } from 'react-router-dom'; // Import useNavigate instead of useHistory
+import React, { useState, useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
 import axios from 'axios';
+//import { toast } from 'react-toastify';
+import { AuthContext } from './AuthContext'; // Import AuthContext
+//import 'react-toastify/dist/ReactToastify.css';
 import '../styles/login.css';
 
 const Login = () => {
+  //toast.configure();
   const [username, setUsername] = useState('');
   const [password, setPassword] = useState('');
-  const navigate = useNavigate(); // Initialize useNavigate hook
+  const navigate = useNavigate();
+  const { login } = useContext(AuthContext); // Use the login function from AuthContext
 
   const handleSubmit = async event => {
     event.preventDefault();
-
-    axios
-      .post('http://localhost:3002/auth/login', {
+    try {
+      const response = await axios.post('http://localhost:3002/auth/login', {
         username: username,
         password: password,
-      })
-      .then(response => {
+      });
+
+      if (response.data.message === 'Invalid username or password') {
+       // toast.error('Invalid username or password');
+      } else {
+        //toast.success('Login successful');
         // Store the token in localStorage
         localStorage.setItem('token', response.data.token);
-        // Store user information in session storage
-        sessionStorage.setItem('user', JSON.stringify(response.data.user));
-        window.postMessage({
-          type: "LOGIN_SUCCESS",
-          token: localStorage.getItem('token'),
-          user: sessionStorage.getItem('user')
-        }, "*");
```

Comment:
@aezouhri This code needs to be put back, in order for the extension login to work, please put it back and re push, If you put it somewhere else let me know where it is so i can double check that the logic works